frontend/src/ImportFromInbus.svelte

```diff
@@ -0,0 +1,197 @@
+    }
+
+
+    async function loadSemesters() {
+        const res = await fetch('/api/semesters')
+        semesters_data = await res.json();
+        parseSemesters();
+    }
+
+
+    async function loadScheduleForSubjectVersionId() {
+        let res = await fetch(`/api/inbus/schedule/subject/version/${subject_inbus_selected.subjectVersionId}`);
+        subject_inbus_schedule = await res.json();
+    }
+
+
+    $: if (subject_inbus_selected) {
+        loadScheduleForSubjectVersionId();
+    }
+
+
+    async function import_activities() {
+        busy = true;
+        const req = {
+            'semester_id': semester,
+            'subject': subject_kelvin_selected,
+            'activities': classes_to_import,
+        };
+
+        const res = await fetch('/api/import/activities', {
+            method: 'POST',
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(req),
+        });
+
+        result = (await res.json());
+        busy = false;
+    }
+
+    loadSemesters();
+    loadInbusAndKelvinSubjects();
+</script>
+
+
+{#if semesters}
+<select bind:value={semester}>
+    {#each semesters as item}
+    <option value="{item.pk}">{item.display}</option>
+    {/each}
+</select>
+{/if}
+
+
+{#if subjects_kelvin}
+<select bind:value={subject_kelvin_selected}>
+    {#each subjects_kelvin as item}
+    <option value={item}>{item.abbr} - {item.name}</option>
+    {/each}
+</select>
+{/if}
+
+
+{#if subjects_inbus_filtered}
+<select bind:value={subject_inbus_selected}>
+    {#each subjects_inbus_filtered as item}
+    <option value={item}>{item.subjectVersionCompleteCode} - {item.subject.abbrev} - {item.subject.title}</option>
+    {/each}
+</select>
+{/if}
+
+
+{#if subject_inbus_schedule}
+<table class="table table-hover table-stripped table-sm">
+    <tbody>
+    {#each subject_inbus_schedule as ca} <!-- `ca` stands for concrete_activity -->
+    <tr>
+        <td>
+            <label>
+                <input type="checkbox" bind:group={classes_to_import} name="classesToImport" value={ca.concreteActivityId} />
+                {ca.educationTypeAbbrev}
+            </label>
+        </td>
+
+        <td>
+            {ca.educationTypeAbbrev}, {ca.educationTypeTitle}
+        </td>
+        <td>
+            {ca.teacherFullNames}
+        </td>
+
+        <td>
+            {ca.weekDayAbbrev}
+        </td>
+
+        <td>
+            {ca.beginTime}
+        </td>
+
+        <td>
+            {ca.endTime}
+        </td>
+    </tr>
+    {/each}
+    </tbody>
+</table>
+{/if}
+
+
+<button class="btn btn-success" on:click={import_activities} disabled={!canImport} class:btn-danger={!canImport}>
+    {#if busy}
+        Importing...
+    {:else}
+        Import
+    {/if}
+</button>
+
+
+{#if result}
+    <table class="table table-sm table-hover table-striped">
+        <tbody>
+        {#each result.users as item}
```

Comment:
I got this error here after importing classes:
```
index.mjs:2374 Uncaught (in promise) Error: {#each} only iterates over array-like objects.
    at validate_each_argument (index.mjs:2374:15)
    at create_if_block$a (ImportFromInbus.svelte:187:28)
    at Object.update [as p] (ImportFromInbus.svelte:184:12)
    at update (index.mjs:1347:36)
    at flush (index.mjs:1307:17)
```
because the import has failed:
```
"Traceback (most recent call last):\n  File \"/projects/vsb/upr/kelvin/common/bulk_import.py\", line 191, in run_inbus\n    class_in_db[c] = Class.objects.get(code=c, semester=semester, subject=subject)\n  File \"/projects/vsb/upr/kelvin/venv/lib/python3.10/site-packages/django/db/models/manager.py\", line 85, in manager_method\n    return getattr(self.get_queryset(), name)(*args, **kwargs)\n  File \"/projects/vsb/upr/kelvin/venv/lib/python3.10/site-packages/django/db/models/query.py\", line 429, in get\n    raise self.model.DoesNotExist(\ncommon.models.Class.DoesNotExist: Class matching query does not exist.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/projects/vsb/upr/kelvin/api/views.py\", line 615, in import_activities\n    res['users'] = list(BulkImport().run_inbus(activities, subject, semester))\n  File \"/projects/vsb/upr/kelvin/common/bulk_import.py\", line 213, in run_inbus\n    class_in_db[c].save()\n  File \"/projects/vsb/upr/kelvin/venv/lib/python3.10/site-packages/django/db/models/base.py\", line 700, in save\n    raise ValueError(\nValueError: save() prohibited to prevent data loss due to unsaved related object 'teacher'.\n"
```

Reply:
Tnaks for the report. I'm able to reproduce a failed import with yet another error :)

Reply:
OK. `User` was not `save`d.